compiler/src/iree/compiler/Modules/Check/IR/CheckOps.td

```diff
@@ -137,26 +137,40 @@ def CHECK_ExpectAlmostEqOp :
     Op<CHECK_Dialect, "expect_almost_eq", [AllTypesMatch<["lhs", "rhs"]>]> {
   let summary = [{Checks that the operands are almost equal}];
   let description = [{
-    Verifies that the buffer view or tensor operands with float elements are
-    almost equal to within an implementation-defined "reasonable" tolerance.
+    Verifies that the buffer view or tensor operands with float elements satisfy
+    the Numpy-style fuzzy-comparision condition with pararameters `atol`,
+    `rtol`, which is the following element-wise on array elements `lhs`, `rhs`:
+    ```
+    abs(lhs - rhs) <= atol + rtol * abs(rhs).
+    ```
 
     Issues a non-fatal failure if the verification fails.
 
+    The `atol`, `rtol` parameters may be omitted, in which case some default
+    value is used. The default `atol` is nonzero, while the default `rtol` is
+    zero, which makes these comparision behave closer to exact comparisons as
+    the values being compared get large.
+
+    This default behavior is supported for legacy compatibility and to support
+    some use cases that legitimately don't care, but the majority of use cases
+    should care and so should provide explicit `atol`, `rtol` values.
+
     ```mlir
-    check.expect_almost_eq(%arg0, %arg1) : tensor<5xf32>
+    check.expect_almost_eq(%arg0, %arg1, atol 1.0e-2, rtol 1.0e-3) : tensor<5xf32>
     ```
   }];
 
   let arguments = (ins
     Optional<HAL_Device>:$device,
     AnyTypeOf<[HAL_BufferView, TensorOf<[AnyFloat]>]>:$lhs,
     AnyTypeOf<[HAL_BufferView, TensorOf<[AnyFloat]>]>:$rhs,
-    F32Attr:$tolerance
+    AnyAttr:$atol,
+    AnyAttr:$rtol
   );
 
   let assemblyFormat = [{
     (`` `<` $device^ `>`)?
-    `` `(` $lhs `,` $rhs `` custom<OptionalFloatTolerance>($tolerance) `)`
+    `` `(` $lhs `,` $rhs `` custom<OptionalFloatTolerance>($atol, $rtol) `)`
```

Comment:
I think you should be able to change to `DefaultValuedAttr` and drop the custom parser/printer for something like this
```
(`` `,` `atol` $atol^)? (`` `,` `rtol` $rtol^)?
```

Not blocking because sometimes I've found the custom parsing/printing finicky around whitespace so let me know if this doesn't work out.

Reply:
Great, thanks, not sure why I got confused earlier and wrote the custom parser. This code had gone through many hoops since last week.